kmp-launchpad-google-utils/src/commonMain/kotlin/com/bottlerocketstudios/launchpad/google/utils/data/network/serviceImpl/airquality/AirQualityApiServiceImpl.kt

```diff
@@ -0,0 +1,75 @@
+package com.bottlerocketstudios.launchpad.google.utils.data.network.serviceImpl.airquality
+
+import com.bottlerocketstudios.launchpad.google.utils.data.model.airquality.AirQualityExceptions
+import com.bottlerocketstudios.launchpad.google.utils.data.model.airquality.CurrentAqiConditionsRequestDto
+import com.bottlerocketstudios.launchpad.google.utils.data.model.airquality.CurrentConditionsResponseDto
+import com.bottlerocketstudios.launchpad.google.utils.data.model.place.LatLngDto
+import com.bottlerocketstudios.launchpad.google.utils.data.network.HttpRoutes
+import com.bottlerocketstudios.launchpad.google.utils.data.network.ktorClient
+import com.bottlerocketstudios.launchpad.google.utils.data.network.service.airquality.AirQualityApiService
+import com.bottlerocketstudios.launchpad.google.utils.data.network.service.apiKey.ApiKeyService
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class AirQualityApiServiceImpl : AirQualityApiService, KoinComponent {
+
+    // DI
+    private val apiKeyService: ApiKeyService by inject()
+    private val client: HttpClient = ktorClient()
+
+    /**
+     * Sets the API key.
+     *
+     * @param apiKey The API key to set.
+     */
+    override fun setApiKey(apiKey: String) {
```

Comment:
Instead of having a "set" function here:

val apiKey:String get() = apiKeyService.getApiKey() - or-
val apiKey: String by lazy { apiKeySrevice.getApiKey() }

Is it needed to be able to externally set the API key?